### PR TITLE
Update AGP to 7 

### PR DESCRIPTION
**Pull request by ShivamPokhriyal**
## Summary

cross-request: https://github.com/dimagi/commcare-core/pull/1034

## Safety Assurance

- [x] If the PR is high risk, "High Risk" label is set
- [x] I have confidence that this PR will not introduce a regression for the reasons below
- [x] Do we need to enhance manual QA test coverage ? If yes, "QA Note" label is set correctly

### CODE REVIEW

**Comment by ShivamPokhriyal** [comment]
Using Java 11
@damagatchi retest this please

**Comment by ShivamPokhriyal** [comment]
@damagatchi retest this please

File: app/build.gradle [hunk 1]
```diff
@@ -82,7 +83,7 @@ dependencies {
     implementation 'androidx.legacy:legacy-support-v4:1.0.0'
     implementation 'androidx.legacy:legacy-preference-v14:1.0.0'
     implementation 'androidx.gridlayout:gridlayout:1.0.0'
-    implementation 'org.bouncycastle:bcprov-jdk15to18:1.69'
+    implementation 'org.bouncycastle:bcprov-jdk15on:1.69'
```
Comment (ShivamPokhriyal) [inline app/build.gradle:86]: Can't use this. From `jdk15on-1.66` Java 16 (major version 59) is required. So we'll have to keep using `jdk15to18`

**Comment by dimagimon** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add labels to this PR</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR does not contain any JIRA issue keys in the PR title or commit messages (e.g. KEY-123)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/AndroidManifest.xml [hunk 2]
```diff
@@ -494,9 +494,6 @@
 
         <activity android:name="org.commcare.activities.CommcareListActivity"/>
 
-        <meta-data android:name="android.content.APP_RESTRICTIONS"
```
Comment (shubham1g5) [inline app/AndroidManifest.xml:497]: is this no longer needed ?

File: app/build.gradle [hunk 3]
```diff
@@ -257,11 +249,6 @@ android {
         exclude 'META-INF/NOTICE.txt'
     }
 
-    dexOptions {
```
Comment (shubham1g5) [inline app/build.gradle:260]: Any comments on why is this removed ?

File: app/build.gradle [hunk 4]
```diff
@@ -415,13 +402,9 @@ android {
...
 
             // grab commcare version from manifest
-            def manifestParser = new DefaultManifestParser(android.sourceSets.main.manifest.srcFile, new BooleanSupplier() {
-                @Override
-                boolean getAsBoolean() {
-                    return true
-                }
-            }, true, null)
-            def ccVersion = manifestParser.getVersionName()
+            // TODO Fix  No signature of method: build_aul2fcabvs6vrsl6qy1mj4j3h.android() is applicable for argument types: (build_aul2fcabvs6vrsl6qy1mj4j3h$_run_closure7) values: [build_aul2fcabvs6vrsl6qy1mj4j3h$_run_closure7@157fe3a8]
```
Comment (shubham1g5) [inline app/build.gradle:405]: ping on this

File: commcare-support-library/build.gradle [hunk 5]
```diff
@@ -43,8 +43,6 @@ android {
     defaultConfig {
         minSdkVersion 16
         targetSdkVersion 30
-        versionCode 1
```
Comment (shubham1g5) [inline commcare-support-library/build.gradle:46]: Is this deprecated ?

File: app/AndroidManifest.xml (hunk 2 above)
Comment (ShivamPokhriyal) [reply app/AndroidManifest.xml:497]: This is a duplicate entry. https://github.com/dimagi/commcare-android/blob/master/app/AndroidManifest.xml#L490 and https://github.com/dimagi/commcare-android/blob/master/app/AndroidManifest.xml#L497. 
So I removed one of them after noticing the warning about the duplicate entries.

File: app/build.gradle (hunk 4 above)
Comment (ShivamPokhriyal) [reply app/build.gradle:405]: Yeah this is something for which I haven't been able to find a replacement for. The documentation for `DefaultManifestParser` doesn't provide any alternative. 
I also haven't seen us use `qabeta` variant in the last 2 years so I'm guessing we can remove it completely?

File: app/build.gradle (hunk 3 above)
Comment (ShivamPokhriyal) [reply app/build.gradle:260]: Ohh, sorry for not adding info on the commit. https://issuetracker.google.com/issues/183295952 this is the related issue tracker which contains more details on the issue. 
```
DSL element 'dexOptions' is obsolete and should be removed.
It will be removed in version 8.0 of the Android Gradle plugin.
Using it has no effect, and the AndroidGradle plugin optimizes dexing automatically.
```

File: commcare-support-library/build.gradle (hunk 5 above)
Comment (ShivamPokhriyal) [reply commcare-support-library/build.gradle:46]: Not really sure why I removed it. I think the library modules don't need the versionCode property, but I can add it back if required?

File: app/build.gradle (hunk 4 above)
Comment (shubham1g5) [reply app/build.gradle:405]: got it, yeah agree that we can remove this variant in another PR.

File: app/build.gradle [hunk 6]
```diff
@@ -523,6 +503,9 @@ android {
 
     // Needed to run espresso tests on release build.
     testBuildType project.ext.TEST_BUILD_TYPE
+    androidResources {
```
Comment (shubham1g5) [inline app/build.gradle:506]: is there a particular advantage of replacing `aaptOptions` with this. Also confirming that you have tested that these files are not making it to the apk with the new code.

File: commcare-support-library/build.gradle (hunk 5 above)
Comment (shubham1g5) [reply commcare-support-library/build.gradle:46]: yeah would be good to add these back just in case if it's getting used in the maven publishing.

File: app/build.gradle (hunk 6 above)
Comment (ShivamPokhriyal) [reply app/build.gradle:506]: I don't exactly remember why I replaced `aaptOptions` with `androidResources`. It was perhaps deprecated but I can't find that information in the docs now. And I didn't test whether the files are present in the apk or not.

File: app/build.gradle (hunk 4 above)
Comment (shubham1g5) [reply app/build.gradle:405]: https://github.com/dimagi/commcare-android/pull/2535

File: commcare-support-library/build.gradle (hunk 5 above)
Comment (shubham1g5) [reply commcare-support-library/build.gradle:46]: going to leave it as it is since the versionName here is anyway wrong

File: app/build.gradle (hunk 6 above)
Comment (shubham1g5) [reply app/build.gradle]: added it back https://github.com/dimagi/commcare-android/pull/2526/commits/df53214f500c691c4f1ae1b8aace4a35e2c107e2 just for the sake of limiting changes. It seems to build fine with it.

**Comment by shubham1g5** [comment]
@damagatchi retest this please

**Review by shubham1g5 — APPROVED** [review]

**Comment by shubham1g5** [comment]
changed jenkins java version to 11
@damagatchi retest this please

**Review by shubham1g5 — APPROVED** [review]
